analyticsclient/client.py

```diff
@@ -91,13 +109,29 @@ def has_resource(self, resource, timeout=None):
 
         """
         try:
-            self._request(resource, timeout=timeout)
+            self._request(METHOD_GET, resource, timeout=timeout)
             return True
         except ClientError:
             return False
 
+    def _get_or_post(self, method, resource, post_data={}, timeout=None, data_format=DF.JSON):
```

Comment:
Nice DRY method!

Defaulting `post_data` to an empty dictionary might end up with unexpected results though, see https://pythonconquerstheuniverse.wordpress.com/category/python-gotchas/.  Easily rectified though!